### PR TITLE
Added Fibonacci spiral and test for it

### PR DESCRIPTION
**Pull request by Atharva-Shah-2298**
## Fibonacci Sphere Method for Sphere Vertices

The `fibonacci_sphere` method is designed for efficiently generating points on the surface of a sphere, such as in the iterations of disperse charges. The time complexity is reduced because of the analytical solution as opposed to an optimization problem solution in disperse charges

### Method Signature:

```python
def fibonacci_sphere(n_points: int, randomize: bool) -> numpy.ndarray of shape (n_points,3)
```

### Example Usage

```python
from dipy.core.sphere import fibonacci_sphere,Sphere
points = fibonacci_sphere(n_points=1000, randomize=True)
sph = Sphere(xyz = points)
```

### CODE REVIEW

File: dipy/core/sphere.py [hunk 1]
```diff
@@ -496,6 +497,44 @@ def disperse_charges(hemi, iters, const=.2):
...
+    """
+    Generate points on the surface of a sphere using Fibonacci Spiral Sampling.
+
+    Parameters
+    ----------
+
+    - n_points (int): The number of points to generate on the sphere surface.
+      Default is 724.
+    - randomize (bool): If True, randomize the starting point on the sphere.
+      Default is True.
```
Comment (skoudoro) [inline dipy/core/sphere.py:510]: Can you remove the default value. 

Also, Can you look at the other function to respect our docstring standard?  currently your docstring does not follow it

File: dipy/core/sphere.py [hunk 2]
```diff
@@ -496,6 +497,44 @@ def disperse_charges(hemi, iters, const=.2):
...
+
+    Returns
+    -------
+
+    array-like (N,3): An array of 3D points representing coordinates
+    on the sphere surface.
```
Comment (skoudoro) [inline dipy/core/sphere.py:516]: same comment as above

File: dipy/core/sphere.py [hunk 3]
```diff
@@ -496,6 +497,44 @@ def disperse_charges(hemi, iters, const=.2):
...
+    offset = 2.0 / n_points
+    increment = np.pi * (3. - np.sqrt(5.))
+
+    for i in range(n_points):
+        y = ((i * offset) - 1) + (offset / 2)
+        r = np.sqrt(1 - pow(y, 2))
+        phi = ((i + random_shift) % n_points) * increment
+        x = np.cos(phi) * r
+        z = np.sin(phi) * r
+        points.append([x, y, z])
```
Comment (skoudoro) [inline dipy/core/sphere.py:533]: this can be done with numpy array and it will be way faster !  I do not see the need of a loop here. so this fucntion can be done without loop. 

Please, can you update ? thanks

**Review by skoudoro — COMMENTED** [review]
Hi @Atharva-Shah-2298,

Thank you for this!  See below an initial review of your PR. Thank you for your future update

**Comment by skoudoro** [comment]
Also, Can you squash your pep8 commit ? thank you

**Review by skoudoro — CHANGES_REQUESTED** [review]
Also, Can you squash your pep8 commit ? thank you

File: dipy/core/sphere.py (hunk 3 above)
Comment (Atharva-Shah-2298) [reply dipy/core/sphere.py]: Now the array is pre populated and updates each point. Thanks for the suggestion

File: dipy/core/sphere.py [hunk 4]
```diff
@@ -496,6 +496,43 @@ def disperse_charges(hemi, iters, const=.2):
...
+        An array of 3D points representing coordinates on the sphere surface.
+
+    """
+    random_shift = 0
+    if randomize:
+        random_generator = np.random.default_rng()
+        random_shift = random_generator.integers(0, n_points)
+
+    indices = np.arange(n_points)
+    offset = 2.0 / n_points
```
Comment (skoudoro) [inline dipy/core/sphere.py:522]: is there a minimal number of points  needed? Can you add a small tests (`if`) to check that and avoid invalid value like a division by zero.

```python
if nb_points < 0:
   raise ValueError("my message")
```

**Review by skoudoro — COMMENTED** [review]
Hi @Atharva-Shah-2298,

Overall, looks good and much better! thank you for the update. 

See below my last comment but it seems ready to be merged.

**Review by skoudoro — APPROVED** [review]
Looks good to me. here the results:


image

with the following code:
```python
In [1]: from dipy.core.sphere import fibonacci_sphere,Sphere

In [2]: points = fibonacci_sphere(n_points=1000, randomize=True)

In [3]: points = fibonacci_sphere(n_points=323, randomize=True)

In [4]: sph = Sphere(xyz = points)

In [5]: from dipy.viz import window, actor

In [6]: scene = window.Scene()
   ...: scene.SetBackground(1, 1, 1)

In [7]: scene.add(actor.point(sph.vertices, window.colors.red, point_radius=0.05))

In [8]: window.show(scene)
```

I will go ahead and merge this PR.

Thank you for this work @Atharva-Shah-2298
